File: apps/web-ui-service/frontend/src/api/qualityEval.ts
import { getJson, postJson } from "../lib/http";

// ── Dataset ─────────────────────────────────────────────────

export interface EvalDatasetItem {
  dataset_id: string;
  project_code: string;
  name: string;
  description: string;
  task_type: string;
  eval_dimensions: string[];
  item_count: number;
  version: number;
  created_at: string | null;
  updated_at: string | null;
}

export interface EvalItemDetail {
  item_id: string;
  requirement_text: string;
  expected_coverage: string[];
  expected_assertions: Record<string, unknown>[];
  expected_page_codes: string[];
  known_issues: string[];
}

export interface EvalDatasetDetail extends EvalDatasetItem {
  items: EvalItemDetail[];
}

export function listEvalDatasets(project = "atp", taskType = ""): Promise<EvalDatasetItem[]> {
  const params = new URLSearchParams({ project });
  if (taskType) params.set("task_type", taskType);
  return getJson<EvalDatasetItem[]>(`/api/quality-eval/datasets?${params.toString()}`);
}

export function getEvalDataset(datasetId: string): Promise<EvalDatasetDetail> {
  return getJson<EvalDatasetDetail>(`/api/quality-eval/datasets/${encodeURIComponent(datasetId)}`);
}

export function createEvalDataset(body: Record<string, unknown>): Promise<EvalDatasetItem> {
  return postJson<EvalDatasetItem>("/api/quality-eval/datasets", body);
}

export function deleteEvalDataset(datasetId: string): Promise<{ status: string }> {
  return postJson<{ status: string }>(`/api/quality-eval/datasets/${encodeURIComponent(datasetId)}`, { _method: "DELETE" });
}

// ── Run ─────────────────────────────────────────────────────

export interface EvalRunItem {
  run_id: string;
  dataset_id: string;
  project_code: string;
  agent_version: string;
  llm_model: string;
  prompt_version: string;
  gate_rule_version: string;
  task_type: string;
  eval_dimensions: string[];
  status: string;
  total_items: number;
  completed_items: number;
  overall_score: number;
  coverage_score: number;
  assertion_score: number;
  executability_score: number;
  consistency_score: number;
  robustness_score: number;
  hallucination_risk: number;
  created_at: string | null;
  finished_at: string | null;
}

export interface EvalResultItem {
  result_id: string;
  run_id: string;
  item_id: string;
  requirement_text: string;
  coverage_score: number;
  coverage_detail: Record<string, unknown>;
  assertion_score: number;
  assertion_detail: Record<string, unknown>;
  executability_score: number;
  executability_detail: Record<string, unknown>;
  consistency_score: number;
  robustness_score: number;
  hallucination_flags: string[];
  hallucination_score: number;
  weighted_score: number;
  latency_ms: number;
  error_message: string;
}

export interface EvalRunReport {
  run_id: string;
  dataset_name: string;
  task_type: string;
  agent_version: string;
  llm_model: string;
  status: string;
  overall_score: number;
  dimension_scores: Record<string, number>;
  issue_breakdown: Record<string, number>;
  compared_to_previous: Record<string, unknown> | null;
  worst_items: Record<string, unknown>[];
}

export function listEvalRuns(project = "atp", datasetId = ""): Promise<EvalRunItem[]> {
  const params = new URLSearchParams({ project });
  if (datasetId) params.set("dataset_id", datasetId);
  return getJson<EvalRunItem[]>(`/api/quality-eval/runs?${params.toString()}`);
}

export function createEvalRun(body: Record<string, unknown>): Promise<EvalRunItem> {
  return postJson<EvalRunItem>("/api/quality-eval/runs", body);
}

export function executeEvalRun(runId: string): Promise<{ status: string; overall_score: number }> {
  return postJson<{ status: string; overall_score: number }>(`/api/quality-eval/runs/${encodeURIComponent(runId)}/execute`, {});
}

export function getEvalRun(runId: string): Promise<EvalRunItem> {
  return getJson<EvalRunItem>(`/api/quality-eval/runs/${encodeURIComponent(runId)}`);
}

export function getEvalRunResults(runId: string): Promise<EvalResultItem[]> {
  return getJson<EvalResultItem[]>(`/api/quality-eval/runs/${encodeURIComponent(runId)}/results`);
}

export function getEvalRunReport(runId: string): Promise<EvalRunReport> {
  return getJson<EvalRunReport>(`/api/quality-eval/runs/${encodeURIComponent(runId)}/report`);
}

export function deleteEvalRun(runId: string): Promise<{ status: string }> {
  return postJson<{ status: string }>(`/api/quality-eval/runs/${encodeURIComponent(runId)}`, { _method: "DELETE" });
}

File: apps/web-ui-service/app/schemas/quality_eval.py
"""AI 测试质量评估中心 — Pydantic Schemas。"""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


# ── Dataset ──────────────────────────────────────────────────

class EvalItemInput(BaseModel):
    requirement_text: str
    expected_coverage: list[str] = Field(default_factory=list)
    expected_assertions: list[dict] = Field(default_factory=list)
    expected_page_codes: list[str] = Field(default_factory=list)
    perturbed_requirement: str = ""
    known_issues: list[str] = Field(default_factory=list)
    category: str = ""
    context_json: dict[str, Any] = Field(default_factory=dict)


class CreateDatasetRequest(BaseModel):
    project_code: str = "atp"
    name: str
    description: str = ""
    task_type: str = "test_case_generation"
    eval_dimensions: list[str] = Field(default_factory=list)
    items: list[EvalItemInput] = Field(default_factory=list)


class DatasetResponse(BaseModel):
    dataset_id: str
    project_code: str
    name: str
    description: str
    task_type: str
    eval_dimensions: list[str]
    item_count: int
    version: int
    created_at: datetime | None = None
    updated_at: datetime | None = None


class DatasetDetailResponse(DatasetResponse):
    items: list[dict[str, Any]] = Field(default_factory=list)


# ── Run ──────────────────────────────────────────────────────

class CreateRunRequest(BaseModel):
    dataset_id: str
    project_code: str = "atp"
    agent_version: str = ""
    llm_model: str = ""
    prompt_version: str = ""
    gate_rule_version: str = ""
    task_type: str = "test_case_generation"
    eval_dimensions: list[str] = Field(default_factory=list)
    repeat_count: int = Field(default=1, ge=1, le=5)


class RunResponse(BaseModel):
    run_id: str
    dataset_id: str
    project_code: str
    agent_version: str
    llm_model: str
    prompt_version: str
    gate_rule_version: str
    task_type: str
    eval_dimensions: list[str]
    status: str
    total_items: int
    completed_items: int
    overall_score: float
    coverage_score: float
    assertion_score: float
    executability_score: float
    consistency_score: float
    robustness_score: float
    hallucination_risk: float
    created_at: datetime | None = None
    finished_at: datetime | None = None


# ── Result ───────────────────────────────────────────────────

class EvalResultResponse(BaseModel):
    result_id: str
    run_id: str
    item_id: str
    requirement_text: str = ""
    coverage_score: float
    coverage_detail: dict[str, Any] = Field(default_factory=dict)
    assertion_score: float
    assertion_detail: dict[str, Any] = Field(default_factory=dict)
    executability_score: float
    executability_detail: dict[str, Any] = Field(default_factory=dict)
    consistency_score: float
    robustness_score: float
    hallucination_flags: list[str] = Field(default_factory=list)
    hallucination_score: float
    weighted_score: float
    latency_ms: int = 0
    error_message: str = ""


# ── Report ───────────────────────────────────────────────────

class RunReportResponse(BaseModel):
    run_id: str
    dataset_name: str
    task_type: str
    agent_version: str
    llm_model: str
    status: str
    overall_score: float
    dimension_scores: dict[str, float] = Field(default_factory=dict)
    issue_breakdown: dict[str, int] = Field(default_factory=dict)
    compared_to_previous: dict[str, Any] | None = None
    worst_items: list[dict[str, Any]] = Field(default_factory=list)